Library/Homebrew/search.rb

```diff
@@ -22,19 +22,29 @@ def self.search_descriptions(string_or_regex, args, search_type: :desc)
 
       if args.formula? || both
         ohai "Formulae"
-        CacheStoreDatabase.use(:descriptions) do |db|
-          cache_store = DescriptionCacheStore.new(db)
-          Descriptions.search(string_or_regex, search_type, cache_store, eval_all).print
+        if eval_all
+          CacheStoreDatabase.use(:descriptions) do |db|
+            cache_store = DescriptionCacheStore.new(db)
+            Descriptions.search(string_or_regex, search_type, cache_store, eval_all).print
+          end
+        else
+          descriptions = Homebrew::API::Formula.all_formulae.transform_values { |data| data["desc"] }
+          Descriptions.search(string_or_regex, search_type, descriptions, eval_all).print
         end
       end
       return if !args.cask? && !both
 
       puts if both
 
       ohai "Casks"
-      CacheStoreDatabase.use(:cask_descriptions) do |db|
-        cache_store = CaskDescriptionCacheStore.new(db)
-        Descriptions.search(string_or_regex, search_type, cache_store, eval_all).print
+      if eval_all
+        CacheStoreDatabase.use(:cask_descriptions) do |db|
+          cache_store = CaskDescriptionCacheStore.new(db)
+          Descriptions.search(string_or_regex, search_type, cache_store, eval_all).print
+        end
+      else
+        descriptions = Homebrew::API::Cask.all_casks.transform_values { |data| data["desc"] }
```

Comment:
I wonder if this warrants printing a warning if there's non-homebrew-cask taps with casks tapped (as they won't be checked)?

Reply:
^see comment on formulae.